### DISCUSSION TITLE
Help restoring a custom field

### DISCUSSION THREAD

**Discussion by F3d0s** [netbox-community/netbox#14310] [Help Wanted!] [answered]
It happened that someone deleted the wrong custom field, which was used for automating a ticket creation when set to true or false. It also holds the date data. I think it was yyyy-mm-dd.

I got an old copy of the DB, from where I could export the custom field in csv and import it to the production environment. But I'm not sure it will work.

Is it possible to restore it with the same name, configuration, and data, or should I restore the VM server to get back the configuration?

**Comment by mtinberg** [comment, marked as answer]
It would probably be easier and safer to restore this into your own local copy of Netbox in a VM, but as long as you had a new dump of the database you can restore, you could restore your old backup db, export the fields you want, then restore the current db, and then re-add your custom_field data to the current records, manually or via an script using the API.  Or looking at the changelog between the old backup and current, and restoring the old db, then re-creating each of the changes that have been made since then, if there aren't too many.

You might also be able to extract the data directly from the text of the DB dump to re-create those changes in the current system.

There is more than one way to do it.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: F3d0s ***@***.***>
Sent: Tuesday, November 21, 2023 7:58 AM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Help restoring a custom field (Discussion #14310)


It happened that someone deleted the wrong custom field, which was used for automating a ticket creation when set to true or false. It also holds the date data. I think it was yyyy-mm-dd.

I got an old copy of the DB, from where I could export the custom field in csv and import it to the production environment. But I'm not sure it will work.

Is it possible to restore it with the same name, configuration, and data, or should I restore the VM server to get back the configuration?

**Reply by F3d0s** [reply]
Following your suggestion, I restored an old DB's backup, then I connected via PGAdmin to it, I navigated into the schema's table (in my case, it was `public.tenancy_tenant`), exported the table into a csv, with Excel, I found and isolated the custom field's name. Then I re-created the custom field in netbox, made the csv readable for importing, and I imported the custom field data that were deleted.

Thank you for your help!
